### PR TITLE
Don't delete provisioning backup when configuration changes

### PR DESCRIPTION
**Pull request by damonbarry**
At startup, if iotedged detects that `config.yaml` has changed it will remove the `{homedir}/cache/` folder, rebuild the `settings_state` file in that folder (the hash used to detect `config.yaml` changes), and remove any running containers so edge agent can recreate them. When this logic was originally introduced into iotedged, the _other_ file in that folder, `provisioning_backup.json`, would have also been recreated a little later in the startup cycle after provisioning took place.

The problem is that, in 1.0.9, the startup sequence was adjusted and provisioning now takes place _before_ we detect changes in `config.yaml`. As a result, `provisioning_backup.json` is created but then subsequently deleted if `config.yaml` changes are detected.

The impact is that users who specify some form of DPS provisioning in their `config.yaml`, and who upgrade their devices to 1.0.9 will not be able to reprovision their offline devices without doing an additional reboot.

This change removes the code that deleted the entire `cache/` folder. The existing code to recreate `settings_state` already creates the file if it didn't exist, or overwrites it otherwise. The provisioning backup file will already have been refreshed earlier in the startup sequence and doesn't need to be recreated.

### CODE REVIEW

File: edgelet/iotedged/src/lib.rs [hunk 1]
```diff
@@ -1357,7 +1357,7 @@ where
     M::Settings: Serialize,
     C: CreateCertificate + GetIssuerAlias + MasterEncryptionKey,
 {
-    // Remove all edge containers and destroy the cache (settings and dps backup)
+    // Remove all edge containers and delete settings state
```
Comment (arsing) [inline edgelet/iotedged/src/lib.rs:1360]: I would remove "delete settings state" from this as well, since there's no explicit call to delete the file.

Comment (damonbarry) [reply edgelet/iotedged/src/lib.rs:1360]: True, will do.

**Comment by damonbarry** [comment]
I'm adding a test right now, btw. It will ensure that after `check_settings_state` is called, any other files in the `cache` directory still remain. Will commit soon...

File: edgelet/iotedged/src/lib.rs (hunk 1 above)
Comment (damonbarry) [reply edgelet/iotedged/src/lib.rs]: Done

**Comment by damonbarry** [comment]
Test added. @arsing can I get another review?

**Comment by damonbarry** [comment]
Sorry @arsing...one more approval? 😟

**Pull request merged**